File: jira/frontend/src/types/epic.ts
// ============================================================================
// Epic Types
// ============================================================================

// ── Enums / Union types ─────────────────────────────────────────────────────

export type EpicStatus = "todo" | "in-progress" | "done";

export const EPIC_COLORS = [
  "#7c3aed", "#2563eb", "#0891b2", "#059669", "#d97706",
  "#dc2626", "#db2777", "#4f46e5", "#0d9488", "#ca8a04",
] as const;

// ── Entity ──────────────────────────────────────────────────────────────────

export interface Epic {
  id: string;
  project_id: string;
  name: string;
  description: string;
  color: string;
  status: EpicStatus;
  start_date: string | null;
  target_date: string | null;
  task_ids: string[];
  tasks_total: number;
  tasks_done: number;
  created_at: string;
  updated_at: string;
}

// ── Form / Request payloads ─────────────────────────────────────────────────

export interface EpicFormData {
  name: string;
  description?: string;
  color: string;
  status: EpicStatus;
  start_date?: string;
  target_date?: string;
}

export type CreateEpicRequest = EpicFormData;
export type UpdateEpicRequest = Partial<EpicFormData>;

File: project-hub/backend/app/schemas/epic.py
from pydantic import BaseModel


class EpicCreate(BaseModel):
    name: str
    description: str = ""
    color: str = "#3b82f6"
    status: str = "todo"
    start_date: str | None = None
    target_date: str | None = None


class EpicUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    color: str | None = None
    status: str | None = None
    start_date: str | None = None
    target_date: str | None = None


class EpicResponse(BaseModel):
    id: str
    project_id: str
    name: str
    description: str
    color: str
    status: str
    start_date: str | None = None
    target_date: str | None = None
    task_ids: list[str] = []
    tasks_total: int = 0
    tasks_done: int = 0
    created_at: str
    updated_at: str

    model_config = {"from_attributes": True}


class LinkTaskRequest(BaseModel):
    task_id: str
